includes/admin/class-wp-job-manager-promoted-jobs-admin.php

```diff
@@ -114,6 +128,54 @@ public function add_action_notice( $actions_handled ) {
 		return $actions_handled;
 	}
 
+	/**
+	 * Add the host of the Promote Job form to the array of allowed redirect hosts.
+	 *
+	 * @param array $hosts Allowed redirect hosts.
+	 * @return array Updated array of allowed redirect hosts.
+	 */
+	public function add_to_allowed_redirect_hosts( $hosts ) {
+		$hosts[] = wp_parse_url( WP_Job_Manager_Helper_API::get_api_base_url(), PHP_URL_HOST );
+		return $hosts;
+	}
+
+	/**
+	 * Handle the action to promote a job listing, validating as well as redirecting to the form on WPJobManager.com.
+	 *
+	 * @return void
+	 */
+	public function handle_promote_job() {
+		$post_id = absint( $_GET['post_id'] ?? 0 );
+		check_admin_referer( self::PROMOTE_JOB_ACTION . '-' . $post_id );
+		if ( ! $post_id ) {
+			wp_die( esc_html__( 'No job listing ID provided for promotion.', 'wp-job-manager' ), '', [ 'back_link' => true ] );
+		}
+		if ( ! is_ssl() ) {
+			wp_die( esc_html__( 'You must be using SSL to promote a job listing.', 'wp-job-manager' ), '', [ 'back_link' => true ] );
+		}
+		if ( ! current_user_can( 'manage_job_listings', $post_id ) || 'job_listing' !== get_post_type( $post_id ) ) {
+			wp_die( esc_html__( 'You do not have permission to promote this job listing.', 'wp-job-manager' ), '', [ 'back_link' => true ] );
+		}
+		$current_user = get_current_user_id();
+		$site_trust   = WP_Job_Manager_Site_Trust_Token::instance();
+		$token        = $site_trust->generate( 'user', $current_user );
+		if ( is_wp_error( $token ) ) {
+			wp_die( esc_html( $token->get_error_message() ) );
+		}
+		$url = add_query_arg(
+			[
+				'user_id'  => $current_user,
+				'job_id'   => $post_id,
+				'token'    => $token,
+				'site_url' => site_url( '', 'https' ),
+				'locale'   => get_user_locale(),
+			],
+			WP_Job_Manager_Helper_API::get_api_base_url() . self::PROMOTE_JOB_FORM_PATH
+		);
+		wp_safe_redirect( $url );
```

Comment:
Should we maybe set something in the job listing to mark that this is in the process of being promoted, maybe? Not sure

Reply:
That's a good call! Should I ping them or do you want to send the question? :thinking:

Reply:
It's something they're working on. In the next few days, they'll send us a plan for that.

BTW, my comment had a wrong word that I just fixed. Sorry if that made the message confusing to understand.